### ISSUE TITLE
Projen coupling to NodeJs vs `bun.sh`

### ISSUE THREAD

**Issue by so0k** [projen/projen#3759]
Does it make sense to support both `Bun` and `NodeJS` as attempted by:

- https://github.com/projen/projen/pull/3491#issue-2214012117

this recent PR:
- https://github.com/projen/projen/pull/3723

couples projen back to `Nodejs` only

**Comment by mrgrain** [comment]
tl;dr projen currently does not have CI infrastructure setup to properly support Bun (or other Node.js alternatives). Someone from the community will need to invest in this. They also need to start thinking about how to decouple JS/TS projects from `NodeProject`.

---

Hey @so0k Thanks for bringing this to my attention. PRs are welcome to try and get this fixed!

The issue is that PRs like #3491 are low-effort fixes (and I really don't mean that in a bad way!) to get using Bun with projen of the ground. While it is unfortunate that `node` was re-introduced in #3723 (and I should have done better!), `node` is very much still used in a lot of places.

Fundamentally to actually support Bun, we lack a few larger pieces of work:
- Decoupling of JS/TS project types from [NodeProject](https://github.com/projen/projen/blob/7014b74830ddcf80a658b5dd4a31a498cd35b85b/src/javascript/node-project.ts#L376C14-L376C25). Needs to start with an RFC.
- Updates to the testing suite to identify and run certain tests with Bun (and probably `node` not being available at all). This is most critical, since otherwise we cannot prevent regressions.
- Remove all usages of `node` to support the above.

**Comment by so0k** [comment]
It was my impression as I tried to create a separate `BunProject extends Project` trying to re-use projen `github`, `build` and `release` namespaced libraries... it seems Bun currently is best used as Package Manager and perhaps runtime for `Apps`, but building and packaging libraries as done here:

https://github.com/projen/projen/blob/e4b9cf0960a29c2e603fee0a657a19e0dceba5df/src/javascript/node-project.ts#L826-L836

I'm not even sure if there's a bun equivalent?

Ref:

- https://github.com/oven-sh/bun/discussions/6034#discussioncomment-7284712

**Comment by mrgrain** [comment]
Yeah, `npm pack` would be one of the scenarios that needs looking at.
